compiler/rustc_middle/src/traits/specialization_graph.rs

```diff
@@ -31,11 +32,23 @@ pub struct Graph {
 
     /// Whether an error was emitted while constructing the graph.
     pub has_errored: bool,
+
+    /// Overlap mode to be used
+    pub overlap_mode: OverlapMode,
 }
 
 impl Graph {
     pub fn new() -> Graph {
-        Graph { parent: Default::default(), children: Default::default(), has_errored: false }
+        Graph {
+            parent: Default::default(),
+            children: Default::default(),
+            has_errored: false,
+            overlap_mode: OverlapMode::Stable,
```

Comment:
I...hmm. I don't know that I would store this as a field, but I guess it's also ok to do so.